### DISCUSSION TITLE
Ability to stream audio from Movie collections!

### DISCUSSION THREAD

**Discussion by cryptokishan** [UnicornsOnLSD/finamp#938] [General]
Everyone might have good collection of movies and listening to them while driving or working is a best feature I can ask. Just like streaming an audio albums, would like to see Finamp expand its ability to streaming audio version of any selected Movie. Think it as an audio book but actual movie instead of narration. I used to listen to movies with tape recorders in 1990s but we lost that ability with newer technologies.

Here is the list of feature list:

- Expand finamp ability to stream audio from a given movie.
- Ability to select a default language of audio from a movie or select a specific version of audio available in the Movie container.
- Ability to offline download of the audio version of movie.

**Comment by Chaphasilor** [comment]
Finamp theoretically can already stream movies, but you have to add them to a playlist. However, unless you actively turn on transcoding, it will stream the entire movie, including the video stream (only audio will be played though). When transcoding, I think the default track will be picked.

**Reply by cryptokishan** [reply]
Thanks, I didn't know that I can add them to playlists. However the app is keep crashing when I play any movie from the list.

**Reply by cryptokishan** [reply]
By the way, I was trying on Jellyfin server 10.10.0 and Finamp Beta on iOS 18.1

**Reply by Chaphasilor** [reply]
Okay, interesting. Did you turn on transcoding?

**Reply by cryptokishan** [reply]
Yes. It is unresponsive first and then crashes.

**Reply by Chaphasilor** [reply]
How large is the file you're trying to play?  
Can you share the logs maybe?

**Reply by cryptokishan** [reply]
Sorry for the late reply. they vary between 1.5 GB to 4 GB depending on movie quality obviously. I tried downloading the audio from the playlist movie but still the same issue. it becomes unresponsive and then crashes when I start playing. I didn't see anything interesting in the app logs, more over request never reach server either.

**Reply by Chaphasilor** [reply]
Can you go into audio settings and reduce the buffer duration to something like 10-30 seconds? By default Finamp buffers 10 minutes of audio, but if it's trying to load up an entire video file that might cause issues. Some phones already run out of RAM for high-quality FLAC 😅

**Reply by cryptokishan** [reply]
no difference. As I indicated earlier, I don't see request reaching Jellyfin Server.

**Reply by Chaphasilor** [reply]
Oh, that's interesting. Can you try sharing the app logs? what is shown in the app is only from the current session, but when you click the share button the full logs file is exported.  
also, there's an issue for this already: https://github.com/jmshrv/finamp/issues/289

**Reply by cryptokishan** [reply]
I think this what I could see from log:

```
[QueueService/FINER] 2024-11-13 11:16:23.674448: Queue (current) [0-1-0-0]: [[null]], {} 
[MusicPlayerBackgroundTask/INFO] 2024-11-13 11:16:23.674510: Shuffle mode changed to AudioServiceShuffleMode.none (false).
[KeepScreenOnHelper/FINE] 2024-11-13 11:16:23.674726: keepingScreenOn: false | mainSetting: While Showing Lyrics | whilePluggedInSetting: true | isPlaying: true | lyricsShowing: false | isPluggedIn: false
[QueueService/FINE] 2024-11-13 11:16:23.674912: Queue is empty
[QueueService/FINEST] 2024-11-13 11:16:23.674976: Saved new rebuilt queue previous:[] current:null seek:null next:[] queue:[]
[QueueService/FINER] 2024-11-13 11:16:23.675017: Queue (current) [0-1-0-0]: [[null]], {} 
[KeepScreenOnHelper/FINE] 2024-11-13 11:16:23.675062: keepingScreenOn: false | mainSetting: While Showing Lyrics | whilePluggedInSetting: true | isPlaying: false | lyricsShowing: false | isPluggedIn: false
[KeepScreenOnHelper/FINE] 2024-11-13 11:16:23.675120: keepingScreenOn: false | mainSetting: While Showing Lyrics | whilePluggedInSetting: true | isPlaying: true | lyricsShowing: false | isPluggedIn: false
[QueueService/FINEST] 2024-11-13 11:16:28.443263: Saved new periodic queue previous:[] current:null seek:0 next:[] queue:[]
[QueueService/SEVERE] 2024-11-13 11:16:40.975697: FormatException: Unsupported itemType
		#0      TabContentType.fromItemType (package:finamp/models/finamp_models.dart:682)
		#1      QueueService.generateMediaItem (package:finamp/services/queue_service.dart:987)
		#2      QueueService._replaceWholeQueue (package:finamp/services/queue_service.dart:492)
		#3      QueueService.startPlayback (package:finamp/services/queue_service.dart:456)
		<asynchronous suspension>
		#4      _SongListTileState.build.<anonymous closure>.<anonymous closure> (package:finamp/components/AlbumScreen/song_list_tile.dart:247)
		<asynchronous suspension>
```

I will watch out for updates on the link you provided. It looks like that what I have been requesting as well.
